### PR TITLE
Feat(QRCode): Enhance 'Scan QR Code From Screen' notifications # #1250

### PR DESCRIPTION
**Pull request by SheathedSharp**
Improve the clarity of notifications in the QR code scanning feature:

1. Permission handling:
   - Show notification when screen recording permission is missing
   - Automatically open system settings for permission grant

feature1

2. Enhanced scanning status notifications: When no QR code found:
   - Title: "Scanned X displays"
   - Subtitle: "No QR codes found"
   - Body: "Try adjusting the QR code position on your screen"

   When invalid QR codes found:
   - Title: "Found X QR code(s)"
   - Subtitle: "No valid Shadowsocks URLs"
   - Body: "QR codes found are not Shadowsocks configuration"

   When valid QR codes found:
   - Title: "Found X Shadowsocks URL(s)"
   - Subtitle: "Scanned X displays, found X QR codes"
   - Body: "Successfully added X server configuration(s)"

feature2

### CODE REVIEW

**Comment by SheathedSharp** [comment]
Resolve invalid QR code scanning issue #1250

**Comment by hiddenSharp429** [comment]
Resolve invalid QR code scanning issue #1250

**Comment by SheathedSharp** [comment]
@lkebin  I saw a recent PR mentioned you, Could you help review this PR?

**Comment by SheathedSharp** [comment]
The following error was encountered during the CI build process:

Makefile.am:21: error: Libtool library used but 'LIBTOOL' is undefined 

This seems to be a configuration issue with the build system and has nothing to do with my code changes. Should I resubmit the PR?

**Review by lkebin — APPROVED** [review]

**Comment by SheathedSharp** [comment]
@lkebin Hi maintainers,Could you please help to re-run the workflow? I noticed the build failed due to some configuration issues, but I don't have the permissions to trigger a rebuild myself.Thank you for your time and code review!

File: .github/workflows/feature.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
       - name: Build
         run: |
           brew install automake
+          brew install autoconf
+          brew install libtool
```
Comment (lkebin) [inline .github/workflows/feature.yml:28]: Could you please add these two rows into release.yaml file too ?

Comment (SheathedSharp) [reply .github/workflows/feature.yml:28]: @lkebin Of course! This is my first time submitting a PR, so I'm still learning the process. Sorry for any inconvenience caused.Thank you for your help and guidance!

Comment (lkebin) [reply .github/workflows/feature.yml:28]: Thank you for your contribution !

**Comment by Mmsbit** [comment]
已经更新到最新版本了，扫描二维码时总是提示未开通权限（但实际已经开通权限）。mac OS端现在无法倒入新街节点了。请问开发及维护者，这个怎么破？

**Comment by SheathedSharp** [comment]
@Mmsbit 试试重启一下应用呢？或者说提供详细的操作细节，方便复现

**Pull request merged**